### ISSUE TITLE
Better document 'index' column in Labels layer properties

### ISSUE THREAD

**Issue by jni** [napari/napari#2596]
## 📚 Documentation

See [this thread on image.sc](https://forum.image.sc/t/visualizing-feature-measurements-in-napari-using-colormaps-as-luts/51567/12)

Currently, if one of the properties columns has title "index", it is assumed to specify which label corresponds to which row of the properties table. This is, however, not documented in the docstring!

Additionally, we could/should consider supporting "label" as the column name (either instead of or in addition to "index"), because that would match the convention for skimage regionprops, which is likely to be a source of such tables. =) And anyway it matches what the layer is, ie "labels".

**Comment by sofroniewn** [comment]
> Additionally, we could/should consider supporting "label" as the column name (either instead of or in addition to "index"), because that would match the convention for skimage regionprops, which is likely to be a source of such tables. =) And anyway it matches what the layer is, ie "labels".

Makes sense to me. I'd support "label" only, with maybe a deprecation warning on "index" until removed. We should do that before updating the docs though if we want to make this change

**Comment by DragaDoncila** [comment]
While doing something else this week I also noticed that our labels tutorial doesn't discuss `properties` or the direct `color` mode at all, so maybe while we're here this is something we should update as well.

**Comment by jluethi** [comment]
Would be awesome to have the direct color mode in the documentation! I only learned about it thanks to the helpful answers from all of you in the forums :)

**Comment by haesleinhuepf** [comment]
> I'd support "label" only, with maybe a deprecation warning on "index" until removed.

I'd like to support @sofroniewn's suggestion here. IMO "label" is the defacto standard here set by [scikit-image regionprops](https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.regionprops_table). We're also using the "label" column in a couple of plugins already.

**Comment by davidpross** [comment]
If you set the label layers features with a Pandas DataFrame that has a named index would it make sense to populate the 'index' column in properties automatically? I was expecting the DataFrame index to be the label lookup 'column'.

**Comment by haesleinhuepf** [comment]
I also stepped into that trap. It becomes tricky when you have the same label over time and multiple measurements over time.

Our use-case is a table with columns "label" and "frame" which technically form the index together.

**Comment by sofroniewn** [comment]
@andy-sweet curious your thoughts here/ how this intersects with any of your current work

**Comment by andy-sweet** [comment]
I think I agree with pretty much everything said here, but let me break it down.

1. Agreed with the initial request of documenting how an `index` is treated as a special case in labels layers. I intend to write [a general guide to properties/features across layer types](https://github.com/napari/docs/issues/23), but I think this belongs in [the existing labels layer tutorial](https://napari.org/howtos/layers/labels.html) as other layers don't have a need for this column.
2. Agreed that we should document the direct color mode in the labels layer, but that should be handled in another issue. We may also want to wait to see if the labels color mode changes as a result of the style encoding work.
3. Agreed that `label` is better column name than `index` for the reasons mentioned above and because `pandas.DataFrame.index` is already reserved.
4. In the spirit of [tidy data](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html) and [the possible future direction of a DataFrame access API](https://data-apis.org/dataframe-protocol/latest/API.html#design-decisions), I'd like to avoid introducing any special handling of `pandas.DataFrame.index`. @haesleinhuepf also makes the good point that in general it's hard to infer what the index means (is it a unique index or a possibly non-unique label). I think that if the `index` or `label` column is documented, then this special handling becomes a little less important too.
